src/blocks/file-blocks/edit/index.tsx

```diff
@@ -0,0 +1,192 @@
+import { FileBlockProps, getLanguageFromFilename } from "@githubnext/utils";
+import { RocketIcon } from "@primer/octicons-react";
+import axios from "axios";
+import { useMemo, useState } from "react";
+import { Hunk, parseDiff } from "react-diff-view";
+import "react-diff-view/style/index.css";
+import SyntaxHighlighter from "react-syntax-highlighter";
+import { diffAsText } from "unidiff";
+import "./index.css";
+
+export default function (props: FileBlockProps) {
+  const { content, context, onRequestUpdateContent } = props;
+
+  const [instruction, setInstruction] = useState<string>("");
+  const [newContent, setNewContent] = useState<string>("");
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const language = getLanguageFromFilename(context.path.split("/").pop() || "");
+
+  const hunks = useMemo(() => {
+    if (!newContent) return [];
+    const hunks = parseDiff(
+      diffAsText(content, newContent, {
+        context: 100,
```

Comment:
you're trying to get the whole file in the diff context here? is there a way to get `diffAsText` to produce a single hunk?

Reply:
it's sadly a pretty restrictive API - this is the best I could do to get all of the unchanged lines in the "diff", so they're at least visible. But I could spend another hour or so poking around and looking at alternative libs!

Reply:
no please do not! I was just curious about the magic number.

Reply:
it's a very valid question! definitely worth more exploration at some point :)